docs/tutorial/from-zero-to-hero-write-your-first-kubernetes-charm/integrate-your-charm-with-postgresql.md

```diff
@@ -359,6 +359,166 @@ This should produce something similar to the output below (of course, with the n
 
 Congratulations, your relation with PostgreSQL is functional!
 
+## Add unit tests for your charm
+
+Now that our charm will extract the database authentication data and endpoints information from the relation data in function `fetch_postgres_relation_data`, we can add a test case to cover this scenario. In the test case, we create a relation and set it in the input state; on the relation-changed event, the database information should be extracted and set as environment variables for our `fastapi-service`:
+
```

Comment:
Maybe we should also add a note here pointing out that we are not testing the `fetch_postgres_relation_data` function directly, we're checking that the response to a Juju event is what it should be? To help people understand the level at which Scenario testing is done.